Presentation/Report.cs

```diff
@@ -69,19 +69,31 @@ public string Url
         {
             get
             {
-                string baseUrl = this.BaseUrl, url = this.Path;
+                string baseUrl = this.BaseUrl, url = this.Path, parameters = string.Empty;
+
+                if (url.Contains("&")) // if url contains Parameters
+                {
+                    int index = url.IndexOf("&"); 
+                    if (index >= 0)
+                    {
+                        parameters = url.Substring(index); // select the parameters
+                        url = url.Substring(0, index);      // strip parametrs away from the original url
+                    }
+                }
 
                 if (baseUrl.EndsWith("/"))
                     baseUrl = baseUrl.Substring(0, baseUrl.Length - 1);
 
                 if (!url.StartsWith("/"))
                     url = "/" + url;
 
-                url = string.Format(
-                    "{0}?{1}&rs:format=IMAGE",
-                    baseUrl,
-                    HttpUtility.UrlEncode(url)
-                    );
+                //url = string.Format(
+                //    "{0}?{1}&rs:format=IMAGE",
+                //    baseUrl,
+                //    HttpUtility.UrlEncode(url)
+                //    );
+
+                url = $"{baseUrl}?{System.Net.WebUtility.UrlEncode(url)}{parameters}&rs:format=IMAGE"; // include the parameters Afetr the URL has been encoded
```

Comment:
To support C# 5 spec, please replace with a compatible string assembly equivalent, such as String.Format, or similar. Thank you.